internal/extramanifest/extramanifest.go

```diff
@@ -179,79 +192,149 @@ func ApplyExtraManifest(ctx context.Context, dc dynamic.Interface, restMapper me
+			mKind := manifest.GetKind()
+			mApiVersion := manifest.GetAPIVersion()
+			mName := manifest.GetName()
+			mNamespace := manifest.GetNamespace()
+
+			// Verify resource
+			if mName == "" {
+				errs = append(errs, fmt.Sprintf("%s resource name is empty", mKind))
+				continue
+			}
 
-	// a map to track all the unknown CRDs in one shot
-	unknownCRDs := make(map[string]string)
-
-	// Apply manifest with dryRun
-	dryRun := true
-
-	// Collect unknownCRDs errors and continue to treat them as part of warning.
-	// Any other error should be treated as real error and return early (which should be priority over warnings)
-	for _, manifest := range manifests {
-		if err := ApplyExtraManifest(ctx, h.DynamicClient, h.Client.RESTMapper(), manifest, dryRun); err != nil {
-			defaultErrMsg := fmt.Sprintf("failed to apply manifest with dryrun: %v", err.Error())
-			var groupDiscoveryErr *discovery.ErrGroupDiscoveryFailed
-			switch {
-			case errors.As(err, &groupDiscoveryErr) || meta.IsNoMatchError(err): // Capture unknown CRD issue
-				key := manifest.GroupVersionKind().String()
-				if _, exists := unknownCRDs[key]; !exists {
-					unknownCRDs[key] = manifest.GetName()
-					h.Log.Info("unknown CRD", "CRD", key)
+			// Verify disallowed resource
+			if kinds, exists := unSupportedResources[mApiVersion]; exists {
+				if lo.Contains(kinds, mKind) {
+					errs = append(errs, fmt.Sprintf("%s[%s] in extramanifests is not allowed", mKind, mName))
+					continue
+				}
+			}
+
+			// Add to namespace set when manifest is Namespace
+			if mKind == "Namespace" {
+				nsManifestsSet[mName] = true
+				continue
+			}
+
+			// Check if manifest's namespace exists on cluster before running dryrun
+			if mNamespace != "" && mNamespace != "default" {
+				ns := &corev1.Namespace{}
+				if err := h.Client.Get(ctx, types.NamespacedName{Name: mNamespace}, ns); err != nil {
+					if k8serrors.IsNotFound(err) {
+						if _, exists := nsManifestsSet[mNamespace]; !exists {
+							// namespace not found on target and not in the configmap, skip dryrun
+							warnings = append(warnings, fmt.Sprintf("%s[%s] - namespace %s not found on cluster or not added in configmap in right order", mKind, mName, mNamespace))
+							continue
+						}
+
+						// namespace not found on target but in the configmap, fake the namespace
+						ns.SetName(mNamespace)
+						if err := h.Client.Create(ctx, ns); err != nil {
+							if !k8serrors.IsAlreadyExists(err) {
+								return validationWarn, fmt.Errorf("failed to create namespace %s: %w", ns, err)
+							}
+						}
+						h.Log.Info(fmt.Sprintf("Created namespace %s for dryrun", ns.Name))
+						fakedNamespacesForDryrun[ns.Name] = true
+					} else {
+						return validationWarn, fmt.Errorf("failed to query namespace %s: %w", ns, err)
+					}
+				}
+			}
+
+			// Apply manifest with dryRun
+			if err := ApplyExtraManifest(ctx, h.DynamicClient, h.Client.RESTMapper(), manifest, dryRun); err != nil {
```

Comment:
```suggestion
			if err := ApplyExtraManifest(ctx, h.DynamicClient, h.Client.RESTMapper(), manifest, true); err != nil {
```
`dryRun` is always true, maybe could be simplified (if so, there's a declaration above that could be removed)